### PR TITLE
FEAT: add private _setup-python action and force cache usage

### PR DESCRIPTION
**Pull request by jorgepiloto**
This pull-request solves #56 by adding a "private" action for setting up Python. This action is then reused across all the different actions in the repository.

### CODE REVIEW

**Comment by RobPasMue** [comment]
Looks like the best solution! Let me review it

File: _release-pypi/action.yml [hunk 1]
```diff
@@ -28,7 +28,7 @@ runs:
   steps:
   
     - name: "Set up Python"
-      uses: actions/setup-python@v4
+      uses: pyansys/actions/_setup-python@main
       with:
```
Comment (RobPasMue) [inline _release-pypi/action.yml:32]: Remember that whenever we do a release, we should change all these "@main" tags in the release branch to point to its corresponding release :)

**Review by RobPasMue — APPROVED** [review]
LGTM!

File: _release-pypi/action.yml (hunk 1 above)
Comment (RobPasMue) [reply _release-pypi/action.yml:32]: lol, didn't know there was a guy named "main" hahaha!

Comment (jorgepiloto) [reply _release-pypi/action.yml:32]: That's right! I noticed this too while working with this. Let's keep that in mind.

**Pull request merged**